### ISSUE TITLE
Training code of BSQ VAE

### ISSUE THREAD

**Issue by Felix-Zhenghao** [FoundationVision/Infinity#80]
Hi, will you release the training code of your BSQ VAE tokenizer? I finetune Infinity on robotics simulation data and found that VAE reconstruction quality is the main bottleneck (perhaps because during VAE training simulation data is not seen).

You said you will launch the training code of VAE in https://github.com/FoundationVision/Infinity/issues/20. Looking forward for that!

**Comment by iFighting** [comment]
@Felix-Zhenghao 
bitwise tokenizer has been sorted out and we are sorting out the code to be compatible with the previous VAR tokenizer

**Comment by Felix-Zhenghao** [comment]
> [@Felix-Zhenghao](https://github.com/Felix-Zhenghao) bitwise tokenizer has been almost already and we are cleaning the code to be compatible with the previous VAR tokenizer

Thank you and looking forward anxiously for that :) Because I know the team has included some special loss as mentioned in https://github.com/FoundationVision/Infinity/issues/51 so I have to wait for the official implementation.

**Comment by ChencongZJU** [comment]
> [@Felix-Zhenghao](https://github.com/Felix-Zhenghao) bitwise tokenizer has been almost already and we are cleaning the code to be compatible with the previous VAR tokenizer

Hi, excited to hear this news. 

I'am trying to reproduce VAR Tokenizer(not the infinity tokenizer). However, using the l2 loss/vq loss/gan loss/lpips loss described in paper, I found a severe codebook collapse during training especially in the former resolution such as 1*1, 3*3. I wonder whether adopt special training trick when training var tokenizer? 

Looking forward to your reply and open-source code. 

Thanks.

**Comment by ChencongZJU** [comment]
> [@Felix-Zhenghao](https://github.com/Felix-Zhenghao) bitwise tokenizer has been almost already and we are cleaning the code to be compatible with the previous VAR tokenizer

@iFighting 
Hi, excited to hear this news.

I'am trying to reproduce VAR Tokenizer(not the infinity tokenizer). However, using the l2 loss/vq loss/gan loss/lpips loss described in paper, I found a severe codebook collapse during training especially in the former resolution such as 11, 33. I wonder whether adopt special training trick when training var tokenizer?

Looking forward to your reply and open-source code.

Thanks.

**Comment by LwkBuaa** [comment]
VAR tokenizer and FSQ tokenizer can both be replicated

**Comment by enjoyyi00** [comment]
@ChencongZJU @Felix-Zhenghao 

Apologies for the delay in open-sourcing the VQ-VAE code—we’ve been working to ensure the implementation is properly cleaned up and documented for public release.

If you need immediate access for research purposes, please refer to the discussion and resources in this issue: [#108](https://github.com/FoundationVision/Infinity/issues/108).

We appreciate your patience and will announce here once the full release is ready. Thank you for your interest in Infinity!
